File: lesson-6/java/Airplane.java
/*
 * The airplane parent class.
 *
 * @author Mohamad Tanbari
 * @version 1.0.0
 * @since 2024-11-05
 */

/**
 * The airplane class
 */
public class Airplane {
    /**
     * The speed property.
     */
    private int speed;

    /**
     * Constructor.
     */
    public Airplane() { }

    /**
     * The speed setter.
     *
     * @param speed The speed to set the airplane to.
     */
    public void setSpeed(int speed) {
        this.speed = speed;
    }

    /**
     * The speed getter.
     *
     * @return Speed as an integer.
     */
    public int getSpeed() {
        return this.speed;
    }
}

File: lesson-6/ts/Airplane.ts
/**
 * This is the airplane class.
 * 
 * Author: Mohamad Tanbari
 * Version: 1.0.0
 * Since: 2024-11-05
 */

/**
 * The Airplane class.
 */
export class Airplane {
  /**
   * The speed property.
   */
  protected _speed: number = 0

  /**
   * The setter for the speed property.
   *
   * @param {number} speed - The speed to set the airplane to.
   */
  public set speed (speed: number) {
    if (speed < 0) {
      this._speed = 0
    } else {
      this._speed = speed
    }
  }

  /**
   * The getter for the speed property.
   *
   * @return {number} speed - The current speed of the airplane.
   */
  public get speed (): number {
    return this._speed
  }
}
